### PR TITLE
Fee and pay aks migration

### PR DESCRIPTION
**Pull request by subrotomukherjee**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+apiVersion: flux.weave.works/v1beta1
+kind: HelmRelease
+metadata:
+  name: ccpay-bulkscanning-api
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml]: ccpay-bulkscanning-api?

File: k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  namespace: fees-pay
+  annotations:
+    flux.weave.works/automated: "true"
+    flux.weave.works/tag.java: glob:prod-*
+spec:
+  releaseName: ccpay-bulkscanning-api
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml]: you might want to change this to ccpay-payment-app

File: k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  rollback:
+    enable: true
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: ccpay-bulkscanning-api
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml]: same here as the above

File: k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    version: 0.0.4
+  values:
+    java:
+      replicas: 2
+      useInterpodAntiAffinity: true
+      applicationInsightsInstrumentKey: "280b07a1-f1a1-4369-8e21-0bb2c2995f1e"
+      image: hmctspublic.azurecr.io/ccpay/bulkscanning-api:prod-*
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml]: This should be 
`hmctspublic.azurecr.io/payment/api:prod-*`

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      enable: true
+    java:
+      replicas: 2
+      image: ${IMAGE_NAME}
+      ingressHost: ${SERVICE_FQDN}
+      applicationInsightsInstrumentKey: "280b07a1-f1a1-4369-8e21-0bb2c2995f1e"
+      environment:
+        SPRING_DATASOURCE_URL: "jdbc:postgresql://payment-postgres-db-aat.postgres.database.azure.com:5432/payment"
+        SPRING_DATASOURCE_USERNAME: payment@payment-postgres-db-aat
+        SPRING_LIQUIBASE_ENABLED: 'true'
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-payment-app.yaml]: This should be disabled. You should use enableDBMigration() step in your jenkinsfile

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: payment-api
+    version: 0.1.0
+  values:
+    rollback:
+      enable: true
+    java:
+      replicas: 2
+      image: ${IMAGE_NAME}
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-payment-app.yaml]: How will it find this ${IMAGE_NAME} ?

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      ingressHost: ${SERVICE_FQDN}
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-payment-app.yaml]: How will it find the ${SERVICE_FQDN}?

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml (hunk 6 above)
Comment (naikajah) [reply k8s/aat/common/ccpay/ccpay-payment-app.yaml]: If you see the file https://github.com/hmcts/cnp-flux-config/blob/master/k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml you will get to know what needs to be done for image name.

`hmctspublic.azurecr.io/payment/api:prod-718aea76`

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml (hunk 7 above)
Comment (naikajah) [reply k8s/aat/common/ccpay/ccpay-payment-app.yaml]: Same here this should be I guess
`payment-api-aat.service.core-compute-aat.internal`

**Comment by naikajah** [comment]
This PR is having conflicts with the bulkscanning app. @subrotomukherjee

File: k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml [hunk 8]
```diff
@@ -33,8 +33,10 @@ spec:
         "ccpay":
           resourceGroup: ccpay
           secrets:
-          - bulkscanning-api-POSTGRES-PASS
-          - AppInsightsInstrumentationKey
+            - bulkscanning-api-POSTGRES-PASS
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-bulkscanning-app.yaml]: These changes shouldn't go with this PR as it is a different application which is working in AAT

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: payment-api
+    version: 0.1.0
+  values:
+    rollback:
+      enable: true
+    java:
+      replicas: 2
+      image: hmctspublic.azurecr.io/ccpay/payment-api:prod-*
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-payment-app.yaml]: This doesn't look right to me. Could you please check if this is correct? IMHO it should be:

`hmctspublic.azurecr.io/payment/api:prod-*`

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+      ingressHost: ccpay-payment-api-aat.service.core-compute-aat.internal
```
Comment (naikajah) [inline k8s/aat/common/ccpay/ccpay-payment-app.yaml]: shouldn't this be payment-api-aat.service.core-compute-aat.internal ?

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: payment-api
+    version: 0.1.0
+  values:
+    rollback:
+      enable: true
+    java:
+      replicas: 2
+      image: hmctspublic.azurecr.io/payment/api:prod-*
```
Comment (adusumillipraveen) [inline k8s/aat/common/ccpay/ccpay-payment-app.yaml]: This should be actual latest prod-tag latest image. not -*

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml [hunk 12]
```diff
@@ -0,0 +1,59 @@
+---
+apiVersion: flux.weave.works/v1beta1
+kind: HelmRelease
+metadata:
+  name: ccpay
```
Comment (adusumillipraveen) [inline k8s/aat/common/ccpay/ccpay-payment-app.yaml]:
```suggestion
  name: ccpay-payment-app
```

File: k8s/aat/common/ccpay/ccpay-payment-app.yaml [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  namespace: fees-pay
+  annotations:
+    flux.weave.works/automated: "true"
+    flux.weave.works/tag.java: glob:prod-*
+spec:
+  releaseName: ccpay
```
Comment (adusumillipraveen) [inline k8s/aat/common/ccpay/ccpay-payment-app.yaml]:
```suggestion
  releaseName: ccpay-payment-api
```

**Pull request merged**